webapi/src/main/twirl/queries/sparql/v1/createNewProject.scala.txt

```diff
@@ -0,0 +1,116 @@
+                      results from the WHERE clause of the update.
+ * @param projectIri the IRI of the new user
+ * @param projectClassIri the IRI of the OWL class that the new project should belong to.
+ * @param shortname the new project's shortname (required / unique).
+ * @param longname the new project's longname.
+ * @param description the new project's description.
+ * @param keywords the new project's keywords.
+ * @param logo the new project's logo.
+ * @param basepath the new project's basepath.
+ * @param isActiveProject the new project's status.
+ * @param hasSelfJoinEnabled the new project's self join status.
+ * @param projectOntologyGraph the new project's ontology graph (required / unique).
+ * @param projectDataGraph the new project's data graph (required / unique).
+ *
+ *@
+@(adminNamedGraphIri: IRI,
+  triplestore: String,
+  projectIri: IRI,
+  projectClassIri: IRI,
+  shortname: String,
+  maybeLongname: Option[String],
+  maybeDescription: Option[String],
+  maybeKeywords: Option[String],
+  maybeLogo: Option[String],
+  basepath: String,
+  status: Boolean,
+  hasSelfJoinEnabled: Boolean,
+  projectOntologyGraph: String,
+  projectDataGraph: String)
+
+prefix xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+
+INSERT {
+    GRAPH ?adminNamedGraphIri {
+        ?projectIri rdf:type ?projectClassIri ;
+
+            knora-base:projectShortname "@shortname"^^xsd:string ;
+
+            @if(maybeLongname.nonEmpty) {
+                knora-base:projectLongname "@maybeLongname.get"^^xsd:string ;
+            }
+
+            @if(maybeDescription.nonEmpty) {
+                knora-base:projectDescription "@maybeDescription.get"^^xsd:string ;
+            }
+
+            @if(maybeKeywords.nonEmpty) {
+                knora-base:keywords "@maybeKeywords.get"^^xsd:string ;
+            }
+
+            @if(maybeLogo.nonEmpty) {
+                knora-base:logo "@maybeLogo.get"^^xsd:string ;
+            }
+
+            knora-base:projectBasepath "@basepath"^^xsd:string ;
+
+            knora-base:status "@status"^^xsd:boolean ;
+
+            knora-base:hasSelfJoinEnabled "@hasSelfJoinEnabled"^^xsd:boolean ;
+
+            knora-base:projectOntologyGraph "@projectOntologyGraph"^^xsd:string ;
+
+            knora-base:projectDataGraph "@projectDataGraph"^^xsd:string .
+    }
+}
+@*
+
+GraphDB's consistency checking requires reasoning, but reasoning interferes with certain things
+in the WHERE clauses of our SPARQL updates, so we set a GraphDB-specific flag to return only
+explicit statements in the WHERE clause here.
+
+*@
+@triplestore match {
+    case "graphdb" | "graphdb-free" => {
+        USING <http://www.ontotext.com/explicit>
+    }
+
+    case other => {}
+}
+WHERE {
```

Comment:
Please add a `FILTER NOT EXISTS` to check that no existing project has the same shortname, and have the Scala code do a SELECT to check whether the update succeeded.